interview-questions.md

```diff
@@ -7,64 +7,64 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument is what the function accepts and will be used to replace some variable inside the function's block of code. A function will have parameters that act as place holders. These parameters have no actual value and can represent any data type the user inputs as arguments. When the function is invoked, the user's input(s) is the argument which can be any data type.
 
-  Researched answer:
+  Researched answer: Arguments are parameter values in a function. They are the real values passed and recieved by the function when they invoke. Not all functions require arguments. However, if a function requires a parameter, missing arguments are set to undefined.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: There are two roles in pair programming: driver and navigator. The driver is the one who uses the mouse and keyboard. They are in charge of writing the code in the text editor. They listen to their partner to guide them through the problem. They may also provide suggestions to help their partner through the problem. The navigator is the one explaining how to solve the problem. It is their job to effectively communicate how to write the code or explain their logic. They can also help by looking up additional information on the internet to assist in finding a solution to their code. Both partners need to communicate with each other, especially when there is a misunderstanding or if either persons feel stuck.
 
-  Researched answer:
+  Researched answer: Pair programming is a type software development technique that requires programmers to work together on one computer. One person is the driver who writes the code and the navigator observes and reviews the lines of code being typed. It is best practice to switch roles to give a chance for both partners to write or observe the code. It allows for collaborative teamwork and increase the quality of the software due to the reduction of possible mistakes that can result in bugs or errors. It requires a lot of communication and is place for both programmers to plan and discuss the code they are working on. In this way, they two partners can clarify or build off on each other's ideas to come up with better solutions. It is a practice that encourages the transfer of knowledge.
 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer: I think it stands for test driven development. You usually want to write your test first before you write your code that has some sort of functionality. As soon as you write the test, you have it fail. Then you write the code that is needed to not make the test fail and then test it.
```

Comment:
This is pretty accurate! I'd maybe mention the methodology name 'Red-Green-Refactor' but I feel like you cover it better in your researched answer. Still great!